src/FS.Physical/PollingFileWatcher.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Threading;
+using Microsoft.Extensions.FileProviders.Physical.Internal;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    internal sealed class PollingFileWatcher : IFileWatcher
+    {
+        public static readonly TimeSpan DefaultPollingInterval = TimeSpan.FromSeconds(4);
+        private readonly string _root;
+        private readonly Timer _timer;
+
+        public PollingFileWatcher(string root, TimeSpan pollingInterval)
+        {
+            _root = root;
+            PollingChangeTokens = new ConcurrentDictionary<string, IPollingChangeToken>(StringComparer.Ordinal);
+
+            // PhysicalFileProvider lazy initializes IFileWatcher instances. It's OK for us to initialize this in the ctor since we anticipate
+            // GetFileChangeToken to be invoked right after.
+            _timer = new Timer(RaiseChangeEvents, state: PollingChangeTokens, dueTime: pollingInterval, period: pollingInterval);
```

Comment:
Don't capture asynclocals onto timer https://github.com/dotnet/corefx/issues/26523#issuecomment-417926130

Reply:
Ok. I port the workaround to 2.2 too